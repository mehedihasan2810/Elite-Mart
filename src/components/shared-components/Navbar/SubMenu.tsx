"use client";

import React from "react";

import Link from "next/link";

interface SubMenuPropsType {
  datasetSubmenu: string;
  className: string;
  onSubmenuLeave: (e: React.PointerEvent<HTMLElement>) => void; //eslint-disable-line
  data: any;
}

const SubMenu = ({
  datasetSubmenu,
  className,
  onSubmenuLeave,
  data,
}: SubMenuPropsType) => {
  console.log(data);
  return (
    <aside
      className={className}
      data-submenu={datasetSubmenu}
      onPointerLeave={onSubmenuLeave}
    >
      {datasetSubmenu === "featured" && (
        <>
          <section>
            <h6>Featured</h6>
            <ul role="list">
              {data.new.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>New For Men</h6>
            <ul role="list">
              {data.new_for_men.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>New For Women</h6>
            <ul role="list">
              {data.new_for_women.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>New For Kids</h6>
            <ul role="list">
              {data.new_for_kids.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
        </>
      )}

      {datasetSubmenu === "men" && (
        <>
          <section>
            <h6>Featured</h6>
            <ul role="list">
              {data.featured.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Shoes</h6>
            <ul role="list">
              {data.shoes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Clothes</h6>
            <ul role="list">
              {data.clothes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
        </>
      )}

      {datasetSubmenu === "women" && (
        <>
          <section>
            <h6>Featured</h6>
            <ul role="list">
              {data.featured.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Shoes</h6>
            <ul role="list">
              {data.shoes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Clothes</h6>
            <ul role="list">
              {data.clothes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
        </>
      )}

      {datasetSubmenu === "kids" && (
        <>
          <section>
            <h6>Featured</h6>
            <ul role="list">
              {data.featured.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Shoes</h6>
            <ul role="list">
              {data.shoes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Clothes</h6>
            <ul role="list">
              {data.clothes.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
        </>
      )}

      {datasetSubmenu === "sale" && (
        <>
          <section>
            <h6>Featured</h6>
            <ul role="list">
              {data.featured.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Mens</h6>
            <ul role="list">
              {data.mens.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Womens</h6>
            <ul role="list">
              {data.womens.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
          <section>
            <h6>Kids</h6>
            <ul role="list">
              {data.kids.map((item: any) => (
                <li key={item.labelId}>
                  <Link href={item.url}>{item.label}</Link>
                </li>
              ))}
            </ul>
          </section>
        </>
      )}
    </aside>
  );
};

export default SubMenu;
