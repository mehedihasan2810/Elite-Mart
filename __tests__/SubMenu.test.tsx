import SubMenu from "@/components/shared-components/Navbar/SubMenu";
import { render, screen } from "@testing-library/react";

const data = {
  new: [
    { labelId: "1", label: "New Arrivals", url: "/products" },
    { labelId: "1", label: "Bestsellers", url: "/products" },
    { labelId: "1", label: "Latest Product", url: "/products" },
    { labelId: "1", label: "Latest Clothing", url: "/products" },
    { labelId: "1", label: "Latest Sneekers", url: "/products" },
  ],
  new_for_men: [
    { labelId: "1", label: "Shoes", url: "/products" },
    { labelId: "1", label: "Clothing", url: "/products" },
    { labelId: "1", label: "Accessories", url: "/products" },
  ],
  new_for_women: [
    { labelId: "1", label: "Shoes", url: "/products" },
    { labelId: "1", label: "Clothing", url: "/products" },
    { labelId: "1", label: "Accessories", url: "/products" },
  ],
  new_for_kids: [
    { labelId: "1", label: "Shoes", url: "/products" },
    { labelId: "1", label: "Clothing", url: "/products" },
    { labelId: "1", label: "Accessories", url: "/products" },
  ],
};

describe("<Products />", () => {
  test("Should render properly & match snapshot", () => {
    // ARRANGE
    const { asFragment } = render(
      <SubMenu
        className=""
        datasetSubmenu="featured"
        data={data}
        onSubmenuLeave={jest.fn()}
      />
    );

    // ASSERT
    expect(asFragment()).toMatchSnapshot();
  });

  test("Featured link should be visible", () => {
    // ARRANGE
    render(
      <SubMenu
        className={""}
        datasetSubmenu="featured"
        data={data}
        onSubmenuLeave={jest.fn()}
      />
    );

    expect(
      screen.getByRole("heading", { name: "New For Men", level: 6 })
    ).toBeInTheDocument();
  });
});
