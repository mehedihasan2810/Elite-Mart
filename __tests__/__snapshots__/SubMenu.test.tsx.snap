// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products /> Should render properly & match snapshot 1`] = `
<DocumentFragment>
  <aside
    class=""
    data-submenu="featured"
  >
    <section>
      <h6>
        Featured
      </h6>
      <ul
        role="list"
      >
        <li>
          <a
            href="/products"
          >
            New Arrivals
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Bestsellers
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Latest Product
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Latest Clothing
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Latest Sneekers
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h6>
        New For Men
      </h6>
      <ul
        role="list"
      >
        <li>
          <a
            href="/products"
          >
            Shoes
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Clothing
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Accessories
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h6>
        New For Women
      </h6>
      <ul
        role="list"
      >
        <li>
          <a
            href="/products"
          >
            Shoes
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Clothing
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Accessories
          </a>
        </li>
      </ul>
    </section>
    <section>
      <h6>
        New For Kids
      </h6>
      <ul
        role="list"
      >
        <li>
          <a
            href="/products"
          >
            Shoes
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Clothing
          </a>
        </li>
        <li>
          <a
            href="/products"
          >
            Accessories
          </a>
        </li>
      </ul>
    </section>
  </aside>
</DocumentFragment>
`;
